lib/order.rb

```diff
@@ -0,0 +1,28 @@
+require_relative 'takeaway'
+require_relative 'menu'
+require_relative 'text'
+
+class Order
+
+attr_reader :basket, :dishes
+
+  def initialize
+    @menu
+    @dishes
+    @basket = Hash.new(0)
+    @item_total
+  end
+
+  def add_to_basket(name, quantity = 1)
+    @basket[name] += quantity
+    return "#{quantity} piece(s) of #{name} is added to the basket."
+  end
+
+  def total_price
+    # @basket[name] + "x" + @basket[quantity] + "= £" + @dishes[name]x@basket[quantity]
```

Comment:
Looks like you were trying to mix up functionality (figuring out the price of multiple items) with presentation (building a string) in a single method - I'm not surprised it was difficult. If you find it hard to build a method, really focus on breaking it down into its smallest parts.